assignment3.txt

```diff
@@ -0,0 +1,6 @@
+Lloyd Page
+1. Yes, however i must be declare and initialized before the for loop
+2. The for loop is an infinite loop, as i goes negative, and will not ever satisfy the closing condition of the for loop. Also, a semicolon after the foor loop results in it doing nothing.
+3.i O(n^2)
+3.ii O(n^2) the third loop evaluates to the infinite sum of 1/n^2 in terms of complexity, which converges to approx 1.644. taking 1.644  over infinity results in practically zero, which means the third for loop is called so rarely that it doesn't matter.
```

Comment:
Hm, not quite! The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2. 

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4)